### ISSUE TITLE
When using version 2.4.6 or above, the error "java.lang.ClassNotFoundException: com.datastax.oss.driver.api.core.tracker.RequestTracker" occurs.

### ISSUE THREAD

**Issue by daiczh** [TongchengOpenSource/smart-doc#546]
When utilizing versions of smart-doc-maven-plugin below 2.4.6, I am able to generate documents in various formats without issues. However, when attempting to produce outputs using version 2.4.6 or higher, the following exception arises:

```
[INFO] Smart-doc Start preparing sources at: 2023-07-14 15:12:03
[INFO] Artifacts that the current project depends on: ["demo-ava"]
[INFO] Smart-doc Starting Create API Documentation at: 2023-07-14 15:12:05
[INFO] API documentation is output to => /data/svn/demo-ava/./src/main/resources/static/doc
[ERROR] 
java.lang.NoClassDefFoundError: com/datastax/oss/driver/api/core/tracker/RequestTracker
    at java.lang.ClassLoader.defineClass1 (Native Method)
    at java.lang.ClassLoader.defineClass (ClassLoader.java:763)
    at java.security.SecureClassLoader.defineClass (SecureClassLoader.java:142)
    at java.net.URLClassLoader.defineClass (URLClassLoader.java:468)
    at java.net.URLClassLoader.access$100 (URLClassLoader.java:74)
    at java.net.URLClassLoader$1.run (URLClassLoader.java:369)
    at java.net.URLClassLoader$1.run (URLClassLoader.java:363)
    at java.security.AccessController.doPrivileged (Native Method)
    at java.net.URLClassLoader.findClass (URLClassLoader.java:362)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:424)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:357)
    at com.power.doc.builder.ProjectDocConfigBuilder.initClassFilesMap (ProjectDocConfigBuilder.java:258)
    at com.power.doc.builder.ProjectDocConfigBuilder.<init> (ProjectDocConfigBuilder.java:108)
    at com.power.doc.builder.HtmlApiDocBuilder.buildApiDoc (HtmlApiDocBuilder.java:88)
    at com.smartdoc.mojo.HtmlMojo.executeMojo (HtmlMojo.java:48)
    at com.smartdoc.mojo.BaseDocsGeneratorMojo.execute (BaseDocsGeneratorMojo.java:165)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
Caused by: java.lang.ClassNotFoundException: com.datastax.oss.driver.api.core.tracker.RequestTracker
    at java.net.URLClassLoader.findClass (URLClassLoader.java:382)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:424)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:357)
    at java.lang.ClassLoader.defineClass1 (Native Method)
    at java.lang.ClassLoader.defineClass (ClassLoader.java:763)
    at java.security.SecureClassLoader.defineClass (SecureClassLoader.java:142)
    at java.net.URLClassLoader.defineClass (URLClassLoader.java:468)
    at java.net.URLClassLoader.access$100 (URLClassLoader.java:74)
    at java.net.URLClassLoader$1.run (URLClassLoader.java:369)
    at java.net.URLClassLoader$1.run (URLClassLoader.java:363)
    at java.security.AccessController.doPrivileged (Native Method)
    at java.net.URLClassLoader.findClass (URLClassLoader.java:362)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:424)
    at java.lang.ClassLoader.loadClass (ClassLoader.java:357)
    at com.power.doc.builder.ProjectDocConfigBuilder.initClassFilesMap (ProjectDocConfigBuilder.java:258)
    at com.power.doc.builder.ProjectDocConfigBuilder.<init> (ProjectDocConfigBuilder.java:108)
    at com.power.doc.builder.HtmlApiDocBuilder.buildApiDoc (HtmlApiDocBuilder.java:88)
    at com.smartdoc.mojo.HtmlMojo.executeMojo (HtmlMojo.java:48)
    at com.smartdoc.mojo.BaseDocsGeneratorMojo.execute (BaseDocsGeneratorMojo.java:165)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at org.codehaus.classworlds.Launcher.main (Launcher.java:47)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  4.610 s
[INFO] Finished at: 2023-07-14T15:12:05+08:00
[INFO] ------------------------------------------------------------------------
```
Related dependency version information:
       Apache Maven 3.6.0
 ```
       <smart-doc.version>2.7.2</smart-doc.version>
        <maven-jar-plugin.version>3.3.0</maven-jar-plugin.version>
        <maven-war-plugin.version>3.4.0</maven-war-plugin.version>
        <maven-deploy-plugin.version>3.1.1</maven-deploy-plugin.version>
        <maven-svn-wagon.version>1.4</maven-svn-wagon.version>
        <maven-source-plugin.version>3.3.0</maven-source-plugin.version>
        <maven-surefire-report-plugin.version>3.1.2</maven-surefire-report-plugin.version>
        <versions-maven-plugin.version>2.16.0</versions-maven-plugin.version>
        <maven.compiler.source>1.8</maven.compiler.source>
        <maven.compiler.target>1.8</maven.compiler.target>
        <java.version>1.8</java.version>
```

Plugin Configuration：
```
            <plugin>
                <groupId>com.github.shalousun</groupId>
                <artifactId>smart-doc-maven-plugin</artifactId>
                <version>${smart-doc.version}</version>
                <configuration>
                    <!--指定生成文档的使用的配置文件,配置文件放在自己的项目中-->
                    <configFile>./src/main/resources/smart-doc.json</configFile>
                </configuration>
            </plugin>

```

**Comment by netdied** [comment]
This is about loading dictionary values from a third-party package, which should not significantly affect the document generation process. In subsequent versions, we will suppress this error.
这个是在load三方包里的字典值，应该不太影响文档的生成。后续版本屏蔽下这个错误。

**Comment by daiczh** [comment]
> This is about loading dictionary values from a third-party package, which should not significantly affect the document generation process. In subsequent versions, we will suppress this error.
这个是在load三方包里的字典值，应该不太影响文档的生成。后续版本屏蔽下这个错误。

Hello, the final document has not been generated.
你好，最终文档没有生成呢~

**Comment by shalousun** [comment]
This is from the package path com.datastax.oss.driver.api.core.tracker.RequestTracker
这个是哪里的com/datastax/oss/driver/api/core/tracker/RequestTracker

**Comment by daiczh** [comment]
> This is from the package path com.datastax.oss.driver.api.core.tracker.RequestTracker
这个是哪里的com/datastax/oss/driver/api/core/tracker/RequestTracker

I'm also curious which JAR file this belongs to or which dependency includes it; I've been searching for a while with no luck...
我也好奇这个是哪个jar里面的或者谁依赖的它，找了半天也没头绪。。。

**Comment by shalousun** [comment]
> >This is from the package path com.datastax.oss.driver.api.core.tracker.RequestTracker
这个时哪里的com/datastax/oss/driver/api/core/tracker/RequestTracker
> 
> I'm also curious which JAR file this belongs to or which dependency includes it; I've been searching for a while with no luck...
我也好奇这个是哪个jar里面的或者谁依赖的它，找了半天也没头绪。。。


Run the command `mvn dependency:tree -DoutputFile=tree.txt` to generate a dependency tree. Search for 'com.datastax' within the output. Then, either exclude the problematic dependency using `<exclusions>` in your `pom.xml`, or include the necessary classes specifically within the plugin configuration if they are required for the documentation generation.

mvn dependency:tree -DoutputFile=tree.txt 打下依赖树，搜下com.datastax，然后exclude剔除下，或者是include在插件里加载下你需要的

**Comment by daiczh** [comment]
1. The Maven dependency tree does not contain 'com.datastax'. I've searched again after encountering the issue, and it's still not present.
2. I've attempted excluding it, but to no avail. I'm unsure if my exclusion syntax is correct: `com.datastax:*`
3. I'm unsure what needs to be included. Specifically, if I just want to document the APIs of the current project and ignore all dependencies, how should I proceed?

1、mvn树里面没有com.datastax，出问题的时候我也搜过，刚刚又打了下看，还是没有
2、exclude我剔除过，还是没用，不知道我的写法对么：com.datastax:*
3、include里面我不知道需要添加哪些，比如我只想打当前项目的api，依赖包全忽略该咋写呢？

**Comment by daiczh** [comment]
Right, this issue emerged when upgrading from version 2.4.5 to 2.4.6. I suspect that in order to address the enum functionality mentioned by the previous user, version 2.4.6 might have introduced additional dependencies that I haven't successfully downloaded.

对了，这个问题是2.4.5升级至2.4.6出现的，所以我在想2.4.6为了解决之前那个大佬说的枚举功能是不是引入了其他依赖，而这个依赖我没下载成功？

**Comment by shalousun** [comment]
The `RequestTracker` class originates from a dependency within your business logic or project.
RequestTracker这个就是来自你业务里面依赖过来的

**Comment by netdied** [comment]
Version 2.4.6 only added support for parsing enum dictionary values and did not introduce any new dependencies. The `RequestTracker` class is related to the Cassandra database connection, and it seems likely that you can exclude it using the configuration:
```
packageExcludeFilters : "com.datastax.*",
```
This should help to bypass the issue.

2.4.6 只添加了对枚举字典值的解析 没有添加新的依赖。
这个是 Cassandra 数据库的链接依赖，应该可以用
packageExcludeFilters :"com.datastax.*",
 进行屏蔽

**Comment by shalousun** [comment]
https://github.com/datastax/java-driver

**Comment by shalousun** [comment]
The exclusion regex should be configured as `com.datastax.oss:.*`; the previously written `com.datastax:*` incorrectly specifies the group at the beginning, and the entire regex does not adhere to the official documentation's guidelines.
exclude正则配置成：com.datastax.oss:.*    前面写的com.datastax:*首先group，整个正则不符合官方文档上写的规范

**Comment by daiczh** [comment]
> The exclusion regex should be configured as `com.datastax.oss:.*`; the previously written `com.datastax:*` incorrectly specifies the group at the beginning, and the entire regex does not adhere to the official documentation's guidelines.
exclude正则配置成：com.datastax.oss:.* 前面写的com.datastax:*首先group，整个正则不符合官方文档上写的规范

Adjusting the regex to `com.datastax.oss:.*`, `com.datastax.*`, or `com.datastax:.*` has not resolved the issue, as the same error persists.
正则改成了`：com.datastax.oss:.*、com.datastax.*、com.datastax:.*`都不行，还是一样的错误。

**Comment by daiczh** [comment]
I managed to resolve the issue by only including my own dependencies. It seems to be related to the introduction of Redis into my project.
我通过仅include我自己的依赖解决了。感觉跟我引入了redis有关系。

**Comment by shalousun** [comment]
> I managed to resolve the issue by only including my own dependencies. It seems to be related to the introduction of Redis into my project.
我通过仅include我自己的依赖解决了。感觉跟我引入了redis有关系。

If you have a clear understanding of which dependencies will be used in generating the documentation, explicitly including them is the optimal approach. This practice of loading only what is necessary for documentation generation also results in the shortest execution time.
如果自己比较清楚生成文档会用到哪些依赖，include是最优的，按需加载生成文档也是耗时最短的

**Comment by shalousun** [comment]
This issue has been resolved in versions following 3.0.2.
